gen_vlen.py

```diff
@@ -0,0 +1,43 @@
+import sys
+import os
+import math
+
+vlen = int(sys.argv[1])
+elen = int(sys.argv[2])
+sail_path = 'model/riscv_vlen.sail'
+
+# Check value of VLEN
+if vlen < 128:
+    vlen = 128
+    print('WARN: VLEN less than 128; setting VLEN to 128\n')
+if vlen % 2 != 0:
+    vlen = vlen - (vlen % 2)
+    print('WARN: VLEN not a power of 2; setting VLEN to %d\n' % vlen)
+
+# Check value of ELEN
+if elen < 8:
+    elen = 8
+    print('WARN: ELEN less than 8; setting ELEN to 8\n')
+if elen % 2 != 0:
+    elen = elen - (elen % 2)
+    print('WARN: ELEN not a power of 2; setting ELEN to %d\n' % elen)
```

Comment:
The values of parameters from the caller should not be considered suggestions, which you can ignore if you don't like their values. They should be considered commands. If the code cannot obey those commands, it should fail immediately with a reasonable error message, instead of doing something other than what was commanded.